libafl/src/bolts/serdeany.rs

```diff
@@ -256,7 +255,20 @@ macro_rules! create_serde_registry_for_trait {
                 where
                     T: $trait_name,
                 {
-                    self.map.insert(unpack_type_id(TypeId::of::<T>()), t);
+                    let id = unpack_type_id(TypeId::of::<T>());
+                    assert!(
```

Comment:
could this be a `debug_assert`? Sounds like runtime overhead
CC @andreafioraldi as well

Reply:
The cost of this should be relatively small, and saves a _lot_ of pain. Not sure if anyone even uses debug mode libafl :thinking: